src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -140,13 +141,16 @@
     private final GlobalTrafficShapingHandler globalTrafficShapingHandler;
 
     ClientToProxyConnection(
+              Channel channel,
             final DefaultHttpProxyServer proxyServer,
             SslEngineSource sslEngineSource,
             boolean authenticateClients,
             ChannelPipeline pipeline,
             GlobalTrafficShapingHandler globalTrafficShapingHandler) {
         super(AWAITING_INITIAL, proxyServer, false);
 
+        this.channel = channel;
```

Comment:
this must be fixed, for some reason default place where this channel is set does not work (look like some sort of race condition)